### PR TITLE
Microsoft.OperationalInsights: Preview version 2023-01-01-preview

### PR DESCRIPTION
**Pull request by dulikvor**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Comment by dulikvor** [comment]
Still pending ARM review.

File: specification/operationalinsights/resource-manager/readme.md [hunk 1]
```diff
@@ -38,6 +38,35 @@ openapi-type: arm
 tag: package-2023-09
 ```
 
+### Tag: package-2023-09
```
Comment (razvanbadea-msft) [inline specification/operationalinsights/resource-manager/readme.md:41]: >package-2023-09 [](http://example.com/codeflow?start=9&length=15)

It does not match with the new version added.

File: specification/operationalinsights/resource-manager/readme.md [hunk 2]
```diff
@@ -38,6 +38,35 @@ openapi-type: arm
...
+
+These settings apply only when `--tag=package-2023-01-preview` is specified on the command line.
+
+```yaml $(tag) == 'package-2023-01-preview'
```
Comment (razvanbadea-msft) [inline specification/operationalinsights/resource-manager/readme.md:45]: >package-2023-01-previe [](http://example.com/codeflow?start=19&length=22)

I see there is a stable version 2023-09-01, shouldn't the preview version be 2024-01-01?

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2023-01-01-preview/SummaryLogs.json [hunk 3]
```diff
@@ -0,0 +1,565 @@
...
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SummaryLogsNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK response definition."
```
Comment (razvanbadea-msft) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2023-01-01-preview/SummaryLogs.json:297]: >OK response definition. [](http://example.com/codeflow?start=28&length=23)

Maybe add a more detailed description for this action

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2023-01-01-preview/SummaryLogs.json [hunk 4]
```diff
@@ -0,0 +1,565 @@
...
+    },
+    "SummaryLogs": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/SummaryLogsProperties",
+          "description": "Summary Logs properties."
+        },
+        "systemData": {
```
Comment (razvanbadea-msft) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2023-01-01-preview/SummaryLogs.json:523]: >systemData [](http://example.com/codeflow?start=9&length=10)

This is redundant as ProxyResource(v5) already has systemData

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2023-01-01-preview/SummaryLogs.json [hunk 5]
```diff
@@ -0,0 +1,565 @@
...
+                "description": "User defined summary log rule."
+              },
+              {
+                "value": "System",
+                "description": "Microsoft builtin summary log rule."
+              }
+            ]
+          }
+        },
+        "isActive": {
```
[the remark this answers is not in the document]
Comment (dulikvor) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2023-01-01-preview/SummaryLogs.json:442]: it's a Boolean also on the RP side. no enum.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2023-01-01-preview/SummaryLogs.json (hunk 4 above)
Comment (dulikvor) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2023-01-01-preview/SummaryLogs.json]: Removed.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2023-01-01-preview/SummaryLogs.json (hunk 3 above)
Comment (dulikvor) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2023-01-01-preview/SummaryLogs.json]: Done.

File: specification/operationalinsights/resource-manager/readme.md (hunk 1 above)
Comment (dulikvor) [reply specification/operationalinsights/resource-manager/readme.md]: Fixed.

**Comment by raych1** [comment]
Approved suppression for Go SDK breaking change since it had been introduced and approved in previous PR.

**Comment by kazrael2119** [comment]
add @JeffreyRichter as the reviewer because this rp mixed preview and stable versions in the default tag

**Comment by dulikvor** [comment]
> add @JeffreyRichter as the reviewer because this rp mixed preview and stable versions in the default tag

Default tag is targeting the latest stable version, not the preview one:
image

**Comment by JeffreyRichter** [comment]
Thanks for adding me. Yes, the preview API (Microsoft.OperationalInsights/preview/2023-01-01-preview/SummaryLogs.json) cannot be part of a GA package.

**Comment by dulikvor** [comment]
> Thanks for adding me. Yes, the preview API (Microsoft.OperationalInsights/preview/2023-01-01-preview/SummaryLogs.json) cannot be part of a GA package.

Hi Jeffrey, it's not part of stable package but a preview one. the change is only scoped to a single child resource reaching public preview. All other resources depicted in the preview package target the latest stable of each resource. if required we can remove all of them only leaving the single relevant resource in public preview. please confirm this is the correct way of method to move forward.

in general, why we do not introduce a new api version for all existing resources?
We limit our api changes to actual resources who needs to undergo the change.
We have hundreds of resources and child resources, dozens of them depicted through manifest.
Introducing a new api version for all is not feasible effort on the RP side and manifest.

**Comment by JeffreyRichter** [comment]
Below is Azure's versioning policy...

By definition, an Azure service is a set of operation endpoints that version uniformly.
The Azure service has a customer-facing name which customers use to find docs, SDKs, etc. 
Each version of the service has its own version of the docs, SDKs, etc. 

A version is either in preview (api-version=YYYY-MM-DD-preview) or in GA (api-version=YYYY-MM-DD) which means that ALL of the endpoints of that version are either all in preview or all in GA.

Whenever a new preview OR GA version of the service (which includes all operation endpoints) ships, ALL previous preview versions should stop working in 90 days.
If an old preview version has some features that did not make GA and are to remain in preview, then a brand new preview version should ship as a superset of the latest GA and this new preview version will continue to have the features that remain in preview. This way, customers using the latest preview are also using the latest GA plus what remains in preview.
The new preview version must have a later date than the latest GA version so customers know that the new preview is based on the latest GA version.
 
The above policy ensures that many related processes work correctly such as SDK Packaging, Azure Customer Support, ​breaking changes, PowerShell/CLI, Portal, Azure Stack and Government Clouds, etc. Deviating from the above policy causes many problems for us and for customers.

@dulikvor : Another option is to separate out your endpoints into multiple services (and SDK packages) so that they can version independently and enter in to preview/GA independently.

**Comment by dulikvor** [comment]
> Below is Azure's versioning policy...
> 
> By definition, an Azure service is a set of operation endpoints that version uniformly. The Azure service has a customer-facing name which customers use to find docs, SDKs, etc. Each version of the service has its own version of the docs, SDKs, etc.
> 
> A version is either in preview (api-version=YYYY-MM-DD-preview) or in GA (api-version=YYYY-MM-DD) which means that ALL of the endpoints of that version are either all in preview or all in GA.
> 
> Whenever a new preview OR GA version of the service (which includes all operation endpoints) ships, ALL previous preview versions should stop working in 90 days. If an old preview version has some features that did not make GA and are to remain in preview, then a brand new preview version should ship as a superset of the latest GA and this new preview version will continue to have the features that remain in preview. This way, customers using the latest preview are also using the latest GA plus what remains in preview. The new preview version must have a later date than the latest GA version so customers know that the new preview is based on the latest GA version.
> 
> The above policy ensures that many related processes work correctly such as SDK Packaging, Azure Customer Support, ​breaking changes, PowerShell/CLI, Portal, Azure Stack and Government Clouds, etc. Deviating from the above policy causes many problems for us and for customers.
> 
> @dulikvor : Another option is to separate out your endpoints into multiple services (and SDK packages) so that they can version independently and enter in to preview/GA independently.

I would like to extend my gratitude for @JeffreyRichter for investing his time assisting our team with tackling the issue at hand. I will close the PR and will re-open a new one once we are able to meet the requirements listed above.

**Pull request closed without merging**